lib/Config/ConfigProcessor.php

```diff
@@ -4,5 +4,10 @@
 
 interface ConfigProcessor
 {
+    /**
+     * @param array<string, mixed> $config
+     *
+     * @return array<string, mixed>
```

Comment:
i wonder if it would be worth adding a type alias for "config" records:

https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

so this could then be `@return config-record` with `@phpstan-import-type config-record from Config` in the class docblock or similar

Reply:
Great idea! I'll try to implement it in next PR:)